### ISSUE TITLE
dhall-format infinite loop

### ISSUE THREAD

**Issue by ocharles** [dhall-lang/dhall-haskell#499]
The following causes `dhall-format` to go into an infinite loop of printing `(`.

```
    let nix-dhall =
          ../../dhall/package.dhall ? https://raw.githubusercontent.com/nix-dhall/nix-dhall/master/dhall/package.dhall 
in let busybox = ../busybox/1.23.2.dhall

in let bootstrap-tools =  nix-dhall.fetchurl
    { name =
        "bootstrap-tools.tar.xz"
    , url =
        "http://tarballs.nixos.org/stdenv-linux/x86_64/4907fc9e8d0d82b28b3c56e3a478a2882f1d700f/bootstrap-tools.tar.xz"
    , sha256 =
        "0p71bij6bfhzyrs8676a8jmpjsfz392s2rg862sdnsk30jpacb43" 
    , executable =
        False
    }
in let DerivationArguments = constructors ../../dhall/Nix/types/DerivationArgument.dhall
in let Builders = constructors (../../dhall/Nix/types/Builder.dhall ../../dhall/Nix/types/Derivation.dhall)
in let Systems = constructors (../../dhall/Nix/types/System.dhall)
in let EnvironmentVariables = constructors (../../dhall/Nix/types/EnvironmentVariable.dhall ../../dhall/Nix/types/Derivation.dhall)
in nix-dhall.derivation (../../dhall/Nix/defaults/DerivationArgs.dhall // { 
  args = [ DerivationArguments.`Text` "ash", DerivationArguments.LocalPath "/home/ollie/work/dhallpkgs/unpack-bootstrap-tools.sh" ]
, builder = Builders.Derivation busybox
, name = "GCC-src"
, system = Systems.x86_64-linux {=}
, environment = [ { name = "tarball", value = EnvironmentVariables.Derivation bootstrap-tools } ]
})
```

**Comment by ocharles** [comment]
It seems to be the `?` syntax. If I remove it from line 2, we're all good.

**Comment by quasicomputational** [comment]
Probably #491, I think.

**Comment by ocharles** [comment]
Oh, it's unfortunate that didn't make 1.15. I'd suggest we get this out as 1.15.1 as it's already caused one person to lose work.

**Comment by Gabriel439** [comment]
This is fixed by #491 

I just opened a pull request to cut a new release in #500
